src/lib/models/keyframe.js

```diff
@@ -6,47 +6,68 @@ const validateSchema = require('jsonschema').validate
 const readFileAsync  = Promise.promisify(require('fs').readFile)
 
 const utils     = require('../shared/utils')
-const git       = require('../shared/git')
-const Component = require('../models/component')
-const Frame     = require('../models/frame')
 
 module.exports = class Keyframe {
-  static fromFile(path) {
-    return readFileAsync(path, 'utf8')
-    .then(this.fromYaml)
-  }
+  static fromFiles(paths) {
+    return Promise.map(paths, path => {
+      return readFileAsync(path, 'utf8').then(utils.parseYaml)
+    })
+    .then((kLayers) => {
+
+      // create and validate core keyframe.
+      var kf = new Keyframe(kLayers[0], true)
+
+      // merge all layers(core keyframe + customizations on top)
+      for (let l of kLayers.slice(1)){
+        kf.mergeLayer(l)
+      }
 
-  static fromYaml(yaml) {
-    return new Keyframe(utils.parseYaml(yaml))
+      // validate
+      var valid = kf.validateKeyframeData()
+      if (valid.errors.length > 0) {
+        for (var e of valid.errors){
+          console.error( 'Error: KeyframeLayer.data.components'+ e.stack.slice(8))
+        }
+        process.exit(1)
```

Comment:
Out of interest why a `process.exit()` instead of a `throw`? Usually it's better to throw than exit. Or does something further up trap it? (Same question with the other instances.)